charts/drax/templates/secret-grafana-slack-contactpoints.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if $.Values.global.alerting.enabled -}}
+{{- if eq $.Values.global.alerting.slackToken "" -}}
+  {{- fail "Error: global.alerting.slackToken must be defined." -}}
+{{- end -}}
+{{- end -}}
+
+{{- $secretName := printf "%s-slack-auth" $.Release.Name -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ $secretName }}
+  annotations:
+    "helm.sh/resource-policy": keep
+type: Opaque
+data:
+  {{- if $.Values.global.alerting.enabled -}}
+  {{- $existingSecret := lookup "v1" "Secret" .Release.Namespace $secretName }}
+  slack-token: {{ get ($existingSecret.data) "slack-token" | default ($.Values.global.alerting.slackToken | b64enc) | quote }}
```

Comment:
The current logic persists the secret in the system. If we install drax once without alerting and then again with alerting, helm will retrieve the old empty token. I think here we need to add a check whether the old token is an empty string or not

Reply:
@dimgogos , Indeed, very good point, can't we just make the secret not persistent?

Reply:
We will have the same issue if the do helm upgrade without uninstalling

Reply:
Something like this should work:

`  {{- if $.Values.global.alerting.enabled -}}
  {{- $existingSecret := lookup "v1" "Secret" .Release.Namespace $secretName }}
  {{- if and ($existingSecret) (get ($existingSecret.data) "slack-token") (ne (get ($existingSecret.data) "slack-token") "") -}}
  slack-token: {{ get ($existingSecret.data) "slack-token" | default ($.Values.global.alerting.slackToken | b64enc) | quote }}
  {{ else }}
    {{- fail "Error: global.alerting.slackToken must be non-empty." -}}
  {{- end }}
`

Reply:
{{- if $.Values.global.alerting.enabled -}}
  {{- $existingSecret := lookup "v1" "Secret" .Release.Namespace $secretName }}
  {{- if $existingSecret -}}
    {{- $slackToken := get $existingSecret.data "slack-token" | b64dec }}
    {{- if eq $slackToken "" -}}
      {{- fail "Error: Existing slack-token in the secret must not be empty." -}}
    {{- end }}
    slack-token: {{ get ($existingSecret.data) "slack-token" | quote }}
  {{- else }}
    slack-token: {{ $.Values.global.alerting.slackToken | b64enc | quote }}
  {{- end }}
{{- else }}
  slack-token: ""
{{- end }}


I tested something like this. They will need to delete the existing secret if it exists and it s empty

Reply:
I updated the code to handle this case